examples/step-69/step-69.cc

```diff
@@ -2562,28 +2574,43 @@ namespace Step69
     ParameterAcceptor::initialize("step-69.prm");
     pcout << "done" << std::endl;
 
-    // Next we create the triangulation:
+    // Next we create the triangulation, assemble all matrices, set up
+    // scratch space, and initialize the DataOut<dim> object:
+
+    {
+      print_head(pcout, "create triangulation");
+      discretization.setup();
 
-    print_head(pcout, "create triangulation");
-    discretization.setup();
+      pcout << "Number of active cells:       "
+            << discretization.triangulation.n_global_active_cells()
+            << std::endl;
 
-    pcout << "Number of active cells:       "
-          << discretization.triangulation.n_global_active_cells() << std::endl;
+      print_head(pcout, "compute offline data");
+      offline_data.setup();
+      offline_data.assemble();
 
-    // Assemble all matrices:
+      pcout << "Number of degrees of freedom: "
+            << offline_data.dof_handler.n_dofs() << std::endl;
 
-    print_head(pcout, "compute offline data");
-    offline_data.setup();
-    offline_data.assemble();
+      print_head(pcout, "set up time step");
+      time_stepping.prepare();
+      schlieren_postprocessor.prepare();
 
-    pcout << "Number of degrees of freedom: "
-          << offline_data.dof_handler.n_dofs() << std::endl;
+      data_out.attach_dof_handler(offline_data.dof_handler);
 
-    // And set up scratch space:
+      constexpr auto problem_dimension =
+        ProblemDescription<dim>::problem_dimension;
+      const auto &component_names = ProblemDescription<dim>::component_names;
 
-    print_head(pcout, "set up time step");
-    time_stepping.prepare();
-    schlieren_postprocessor.prepare();
+      for (unsigned int i = 0; i < problem_dimension; ++i)
+        {
+          output_vector[i].reinit(offline_data.partitioner);
+          data_out.add_data_vector(output_vector[i], component_names[i]);
```

Comment:
I read the first line as just resizing `output_vector[i]`. I guess it also copies, but that wasn't clear to me. Can we make that more obvious? (If necessary through a comment.)

Reply:
Wait, I must have misunderstood this. Are you saying that you're pro-actively attach these vectors to the `DataOut` object, and only later fill the vector? That might work, but I'm not sure that was ever the intention of the class to work that way. I've always thought that it could also copy these vectors when they are handed to the class.